protocols/meson.build

```diff
@@ -0,0 +1,20 @@
+wayland_scanner = find_program('wayland-scanner')
+wayland_protos_dep = dependency('wayland-protocols')
+wl_protocol_dir = wayland_protos_dep.get_pkgconfig_variable('pkgdatadir')
+wayland_scanner_code = generator(
+	wayland_scanner,
+	output: '@BASENAME@-protocol.c',
+	arguments: ['private-code', '@INPUT@', '@OUTPUT@'])
+wayland_scanner_client = generator(
+	wayland_scanner,
+	output: '@BASENAME@-protocol.h',
+	arguments: ['server-header', '@INPUT@', '@OUTPUT@'])
```

Comment:
You might also be interested in https://mesonbuild.com/Wayland-module.html

Reply:
I'm not sure this is possible, the wlroots includes require that the protocol file is named `@BASENAME@-protocol.h`, and the module you mentioned seems to only support naming it as `@BASENAME@-server-protocol.h`

Reply:
Hmm, I see.

That's probably a simple matter of a tiny source code modification, but OTOH that *is* more than a build system switch.

Reply:
Well, the source code that requires this is brought in from `/usr/include/`, so that kinda complicates things

Reply:
Oh huh, so it is.

This is probably the fault of https://github.com/mesonbuild/meson/pull/9985 which always calls the header `{name}-{side}-protocol.h`, even though it looks like the wayland intention is for `{name}-protocol.h` for the server and `{name}-client-protocol.h` for the client? Maybe this needs to be fixed.